opm/autodiff/fastSparseOperations.hpp

```diff
@@ -177,7 +177,89 @@ inline void fastSparseDiagProduct(const Eigen::SparseMatrix<double>& lhs,
     }
 }
 
+template<typename Lhs, typename Rhs>
+inline bool
+equalSparsityPattern(const Lhs& lhs, const Rhs& rhs)
+{
+  // if both matrices have equal storage and non zeros match, we can check sparsity pattern
+  bool equal = (Lhs::IsRowMajor == Rhs::IsRowMajor) && (lhs.nonZeros() == rhs.nonZeros());
+
+  // check complete sparsity pattern
+  if( equal )
+  {
+    typedef typename Eigen::internal::remove_all<Lhs>::type::Index Index;
+    const Index nnz = lhs.nonZeros();
 
+    const Index* rhsJ = rhs.innerIndexPtr();
+    const Index* lhsJ = lhs.innerIndexPtr();
+
+    for(Index i=0; i<nnz; ++i )
```

Comment:
I think the inner indices would be 0, 1, 1, 2 for both matrices if they are column major, right?